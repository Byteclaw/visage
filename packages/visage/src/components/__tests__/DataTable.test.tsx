import React from 'react';
import { createTestTheme, render } from './render';
import {
  DataTable,
  DataTableBody,
  DataTableColumn,
  DataTableColumnSkeleton,
  DataTableFooter,
  DataTableFooterRow,
  DataTableHeader,
  DataTableHeaderColumn,
  DataTableHeaderRow,
  DataTableNumericColumnSkeleton,
  DataTableRow,
} from '../DataTable';

describe('DataTable', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<DataTable />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTable face', () => {
    const { asFragment } = render(<DataTable />, {
      ds: {
        theme: createTestTheme({
          DataTable: {
            color: 'black',
          },
        }),
      },
    });

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableBody', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <DataTableBody />
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableBody face', () => {
    const { asFragment } = render(
      <table>
        <DataTableBody />
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableBody: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableColumn', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <tr>
            <DataTableColumn />
          </tr>
        </tbody>
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableColumn face', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <tr>
            <DataTableColumn />
          </tr>
        </tbody>
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableColumn: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableColumnSkeleton', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <tr>
            <DataTableColumnSkeleton />
          </tr>
        </tbody>
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableColumnSkeleton face', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <tr>
            <DataTableColumnSkeleton />
          </tr>
        </tbody>
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableColumnSkeleton: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableFooter', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <DataTableFooter />
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableFooter face', () => {
    const { asFragment } = render(
      <table>
        <DataTableFooter />
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableFooter: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableFooterRow', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <tfoot>
          <DataTableFooterRow />
        </tfoot>
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableFooterRow face', () => {
    const { asFragment } = render(
      <table>
        <tfoot>
          <DataTableFooterRow />
        </tfoot>
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableFooterRow: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableHeader', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <DataTableHeader />
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableHeader face', () => {
    const { asFragment } = render(
      <table>
        <DataTableHeader />
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableHeader: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableHeaderColumn', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <thead>
          <tr>
            <DataTableHeaderColumn />
          </tr>
        </thead>
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableHeaderColumn face', () => {
    const { asFragment } = render(
      <table>
        <thead>
          <tr>
            <DataTableHeaderColumn />
          </tr>
        </thead>
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableHeaderColumn: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableHeaderRow', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <thead>
          <DataTableHeaderRow />
        </thead>
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableHeaderRow face', () => {
    const { asFragment } = render(
      <table>
        <thead>
          <DataTableHeaderRow />
        </thead>
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableHeaderRow: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableNumericColumnSkeleton', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <tr>
            <DataTableNumericColumnSkeleton />
          </tr>
        </tbody>
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableNumericColumnSkeleton face', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <tr>
            <DataTableNumericColumnSkeleton />
          </tr>
        </tbody>
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableNumericColumnSkeleton: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('DataTableRow', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <DataTableRow />
        </tbody>
      </table>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('is extendable by DataTableRow face', () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <DataTableRow />
        </tbody>
      </table>,
      {
        ds: {
          theme: createTestTheme({
            DataTableRow: {
              color: 'black',
            },
          }),
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
