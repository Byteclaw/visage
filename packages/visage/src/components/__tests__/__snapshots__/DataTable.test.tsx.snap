// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable is extendable by DataTable face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
  color: rgb(0,0,0);
}

.emotion-0 th,
.emotion-0 td {
  vertical-align: top;
}

<table
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`DataTable renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  border-collapse: collapse;
  border-spacing: 0;
  width: 100%;
}

.emotion-0 th,
.emotion-0 td {
  vertical-align: top;
}

<table
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`DataTableBody is extendable by DataTableBody face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  color: rgb(0,0,0);
}

<table>
    <tbody
      class="emotion-0"
    />
  </table>
</DocumentFragment>
`;

exports[`DataTableBody renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  <table>
    <tbody
      class="emotion-0"
    />
  </table>
</DocumentFragment>
`;

exports[`DataTableColumn is extendable by DataTableColumn face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  padding: 16px;
  text-align: left;
  color: rgb(0,0,0);
}

<table>
    <tbody>
      <tr>
        <td
          class="emotion-0"
          data-numeric="false"
          data-total="false"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`DataTableColumn renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  padding: 16px;
  text-align: left;
}

<table>
    <tbody>
      <tr>
        <td
          class="emotion-0"
          data-numeric="false"
          data-total="false"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`DataTableColumnSkeleton is extendable by DataTableColumnSkeleton face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  color: rgb(0,0,0);
}

<table>
    <tbody>
      <tr>
        <td
          class="emotion-0"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`DataTableColumnSkeleton renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  <table>
    <tbody>
      <tr>
        <td
          class="emotion-0"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`DataTableFooter is extendable by DataTableFooter face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  color: rgb(0,0,0);
}

<table>
    <tfoot
      class="emotion-0"
    />
  </table>
</DocumentFragment>
`;

exports[`DataTableFooter renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  <table>
    <tfoot
      class="emotion-0"
    />
  </table>
</DocumentFragment>
`;

exports[`DataTableFooterRow is extendable by DataTableFooterRow face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  color: rgb(0,0,0);
}

<table>
    <tfoot>
      <tr
        class="emotion-0"
      />
    </tfoot>
  </table>
</DocumentFragment>
`;

exports[`DataTableFooterRow renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  <table>
    <tfoot>
      <tr
        class="emotion-0"
      />
    </tfoot>
  </table>
</DocumentFragment>
`;

exports[`DataTableHeader is extendable by DataTableHeader face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  color: rgb(0,0,0);
}

.emotion-0 th {
  border-bottom-style: solid;
  border-bottom-color: rgb(184,184,184);
  border-bottom-width: 1px;
}

<table>
    <thead
      class="emotion-0"
    />
  </table>
</DocumentFragment>
`;

exports[`DataTableHeader renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 th {
  border-bottom-style: solid;
  border-bottom-color: rgb(184,184,184);
  border-bottom-width: 1px;
}

<table>
    <thead
      class="emotion-0"
    />
  </table>
</DocumentFragment>
`;

exports[`DataTableHeaderColumn is extendable by DataTableHeaderColumn face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  font-weight: 600;
  padding: 16px;
  text-align: left;
  color: rgb(0,0,0);
}

<table>
    <thead>
      <tr>
        <th
          class="emotion-0"
          data-numeric="false"
          data-total="false"
        />
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;

exports[`DataTableHeaderColumn renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  font-weight: 600;
  padding: 16px;
  text-align: left;
}

<table>
    <thead>
      <tr>
        <th
          class="emotion-0"
          data-numeric="false"
          data-total="false"
        />
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;

exports[`DataTableHeaderRow is extendable by DataTableHeaderRow face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  color: rgb(0,0,0);
}

<table>
    <thead>
      <tr
        class="emotion-0"
      />
    </thead>
  </table>
</DocumentFragment>
`;

exports[`DataTableHeaderRow renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  <table>
    <thead>
      <tr
        class="emotion-0"
      />
    </thead>
  </table>
</DocumentFragment>
`;

exports[`DataTableNumericColumnSkeleton is extendable by DataTableNumericColumnSkeleton face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  color: rgb(0,0,0);
}

<table>
    <tbody>
      <tr>
        <td
          class="emotion-0"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`DataTableNumericColumnSkeleton renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  <table>
    <tbody>
      <tr>
        <td
          class="emotion-0"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`DataTableRow is extendable by DataTableRow face 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0 {
  color: rgb(0,0,0);
}

.emotion-0:hover td {
  color: rgb(255,255,255);
  background-color: rgb(184,184,184);
}

<table>
    <tbody>
      <tr
        class="emotion-0"
      />
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`DataTableRow renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 600;
}

<div
    aria-live="polite"
    class="emotion-0"
    data-toastmanager="true"
  />
  .emotion-0:hover td {
  color: rgb(255,255,255);
  background-color: rgb(184,184,184);
}

<table>
    <tbody>
      <tr
        class="emotion-0"
      />
    </tbody>
  </table>
</DocumentFragment>
`;
